solr/core/src/java/org/apache/solr/handler/MoreLikeThisHandler.java

```diff
@@ -395,7 +395,7 @@ private BooleanQuery getBoostedQuery(Query mltquery) {
 
     public DocListAndSet getMoreLikeThis(
         int id, int start, int rows, List<Query> filters, int flags) throws IOException {
-      Document doc = reader.document(id);
+      Document doc = reader.storedFields().document(id);
```

Comment:
Creation of StoredFields can be pulled into the `MoreLikeThisHandler.MoreLikeThisHelper` ctor (it's only ever created as a local variable/instance, so will by definition be thread-safe). Maybe annotate `MoreLikeThisHandler.MoreLikeThisHelper` class as NotThreadSafe to guard against accidental future misuse?

Reply:
@vinayakphegde I think this was missed being changed to `this.storedFields`?